cmake/onnxruntime.cmake

```diff
@@ -144,40 +149,50 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Android" AND onnxruntime_BUILD_JAVA)
   # copy the header files one by one
   foreach(h_ ${ANDROID_AAR_HEADERS})
     get_filename_component(HEADER_NAME_ ${h_} NAME)
-    configure_file(${h_} ${ANDROID_HEADERS_DIR}/${HEADER_NAME_} COPYONLY)
+    add_custom_command(TARGET onnxruntime POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_if_different ${h_} ${ANDROID_HEADERS_DIR}/${HEADER_NAME_})
   endforeach()
 endif()
 
-target_link_libraries(onnxruntime PRIVATE
-    onnxruntime_session
-    ${onnxruntime_libs}
-    ${PROVIDERS_ACL}
-    ${PROVIDERS_ARMNN}
-    ${PROVIDERS_COREML}
-    ${PROVIDERS_DML}
-    ${PROVIDERS_MIGRAPHX}
-    ${PROVIDERS_NNAPI}
-    ${PROVIDERS_NUPHAR}
-    ${PROVIDERS_RKNPU}
-    ${PROVIDERS_ROCM}
-    ${PROVIDERS_VITISAI}
-    ${PROVIDERS_INTERNAL_TESTING}
-    ${onnxruntime_winml}
-    onnxruntime_optimizer
-    onnxruntime_providers
-    onnxruntime_util
-    ${onnxruntime_tvm_libs}
-    onnxruntime_framework
-    onnxruntime_graph
-    onnxruntime_common
-    onnxruntime_mlas
-    onnxruntime_flatbuffers
-    ${onnxruntime_EXTERNAL_LIBRARIES})
+set(onnxruntime_INTERNAL_LIBRARIES
+  onnxruntime_session
+  ${onnxruntime_libs}
+  ${PROVIDERS_ACL}
+  ${PROVIDERS_ARMNN}
+  ${PROVIDERS_COREML}
+  ${PROVIDERS_DML}
+  ${PROVIDERS_MIGRAPHX}
+  ${PROVIDERS_NNAPI}
+  ${PROVIDERS_NUPHAR}
+  ${PROVIDERS_RKNPU}
+  ${PROVIDERS_ROCM}
+  ${PROVIDERS_VITISAI}
+  ${PROVIDERS_INTERNAL_TESTING}
+  ${onnxruntime_winml}
+  onnxruntime_optimizer
+  onnxruntime_providers
+  onnxruntime_util
+  ${onnxruntime_tvm_libs}
+  onnxruntime_framework
+  onnxruntime_graph
+  onnxruntime_common
+  onnxruntime_mlas
+  onnxruntime_flatbuffers
+)
 
 if (onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS)
-  target_link_libraries(onnxruntime PRIVATE onnxruntime_language_interop onnxruntime_pyop)
+  list(APPEND onnxruntime_INTERNAL_LIBRARIES
+    onnxruntime_language_interop
+    onnxruntime_pyop
+  )
 endif()
 
+# If you are linking a new library, please add it to the list onnxruntime_INTERNAL_LIBRARIES or onnxruntime_EXTERNAL_LIBRARIES,
+# Please do not add a library directly to the target_link_libraries command
```

Comment:
is this restriction just for the onnxruntime target?

Reply:
Other target could use the same restriction, but it would be too much to add the notes to every target

Reply:
```suggestion
# Please do not add a library directly to this target_link_libraries command
```